packages/base/src/MaterialComponent.ts

```diff
@@ -76,23 +76,33 @@ export abstract class MaterialComponent<
     }
 
     const element = this.materialDom(componentProps);
-    element.attributes = element.attributes || {};
-
-    element.attributes.className = `${userDefinedClasses} ${this.getClassName(
+    let propName = 'attributes';
```

Comment:
Suggestion: throw this into a utility to avoid repeated lookups and dynamic property access:

```js
import { h } from 'preact';

const isX = 'props' in h('a');

export const getProps = isX ? vnode => vnode.props : vnode => vnode.attributes;

export const setProps = isX ? (vnode, props) => vnode.props = props : (vnode, props) => vnode.attributes = props;
```

Reply:
(see https://gist.github.com/developit/17fdda2e2236d0c390e8a9bd947f044d)